File: LeetCode/Medium/0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.go
func findDuplicates(nums []int) []int {
    n := len(nums)
    ret := make([]int, 0)
    exist := make([]bool, n + 1)

    for _, v := range nums {
        if exist[v] {
            ret = append(ret, v)
        } else { exist[v] = true }
    }

    return ret
}

File: LeetCode/Medium/0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        int n = nums.length + 1;
        List<Integer> ret = new ArrayList();
        boolean[] exist = new boolean[n+1];
        
        for (int v : nums){
            if (exist[v]) ret.add(v);
            else exist[v] = true;
        }
         
        return ret;
    }
}